roombasim/pittras/task/hold_position_task.py

```diff
@@ -0,0 +1,45 @@
+'''
+hold_position_task.py
+'''
+
+import numpy as np
+
+from roombasim.ai import Task
+
+class HoldPositionTaskStates:
+    '''
+    States for the Hold Position task.
+    '''
+    init = 0
+    holding = 1
+    done = 2
+
+class HoldPositionTask(Task):
+    '''
+    A task to hold the drone at its current absolute position for some duration
+    of time. The task accepts a float64 that specifies such duration in seconds,
+    which means holding the current position indefinitely if less than or equal
+    to zero.
+    '''
+
+    def __init__(self, hold_duration):
+        self.hold_duration = hold_duration
+        self.state = HoldPositionTaskStates.init
+        self.start_time = 0
+
+    def update(self, delta, elapsed, state_controller, environment):
+        if self.done == HoldPosition.done:
+            # TODO(zacyu): Signal the termination of the task when such
+            #              capability is added to the framework.
+            return
+
+        if self.state == HoldPositionTaskStates.init:
+            # TODO(zacyu): Remove this when `elapsed` is changed to represent
+            #              the elipsed time since the start of the task.
+            self.start_time = elapsed
+            self.state = HoldPositionTaskStates.holding
```

Comment:
Should this task ensure that the velocity of the drone is set to zero?

I don't know if this should be done like the XYZ Translation task where it uses acceleration to move the drone or set the velocity to zero.

Reply:
Right, I should have considered that. The only concern is that velocity doesn't change immediately and setting acceleration would involve using negative velocity to move the drone back to its initial absolution position, a similar logic to what is in XYZ Translation. I am not sure what would be the best approach here.

Reply:
What you could do is "capture" the position of the drone when the hold position task is created and use the same PID controller logic as the XYZTranslateTask to move to that position.

Reply:
I "captured" the position upon the first `update` :) Please let me know if there is a way to do it in `__init__`.